nicegui/elements/button.py

```diff
@@ -35,7 +35,11 @@ def __init__(self,
             self._props['icon'] = icon
 
         if on_click:
-            self.on('click', lambda _: handle_event(on_click, ClickEventArguments(sender=self, client=self.client)), [])
+            self.on_click(on_click)
+
+    def on_click(self, callback: Callable[..., Any]) -> None:
+        """Set a callback to be invoked when the button is clicked."""
+        self.on('click', lambda _: handle_event(callback, ClickEventArguments(sender=self, client=self.client)), [])
```

Comment:
i think i'd appreciate if this were the actual builder pattern, and we did

```suggestion
    def on_click(self, callback: Callable[..., Any]) -> Self:
        """Set a callback to be invoked when the button is clicked."""
        self.on('click', lambda _: handle_event(callback, ClickEventArguments(sender=self, client=self.client)), [])
        return self
```

Reply:
Oh sure, I forgot to return `self`. 👍